### PR TITLE
[WT] Audiogram parsing - PoC

### PR DESCRIPTION
**Pull request by wojciechtrapkowski**
This PR introduces a **proof of concept** for the hearing loss prediction pipeline. It includes basic modularization of the code, with initial functionality working end-to-end on sample data.

>⚠️ **Note:** This implementation is still **work in progress**.  Further work is required.

### CODE REVIEW

**Review by Grochman — APPROVED** [review]
Very elegant start, good foundation

File: input_data.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import pandas as pd
+import numpy as np
+
+
+def load_data(filepath):
+    df = pd.read_csv(filepath)
+    X_raw = df.iloc[:, 0:14].values
+    y = df.iloc[:, 14:].values
+    return X_raw, y
```
Comment (tarnowsky) [inline input_data.py]: #### Consider adding error handling

The load_data function lacks error handling for common issues like missing files, malformed CSV data, or insufficient columns.

```suggestion
def load_data(filepath):
    try:
        df = pd.read_csv(filepath)
        if df.shape[1] < 14:
            raise ValueError(f"CSV must have at least 14 columns, found {df.shape[1]}")
        X_raw = df.iloc[:, 0:14].values
        y = df.iloc[:, 14:].values
        return X_raw, y
    except FileNotFoundError:
        raise FileNotFoundError(f"Data file not found: {filepath}")
    except pd.errors.EmptyDataError:
        raise ValueError("CSV file is empty")
    except Exception as e:
        raise RuntimeError(f"Error loading data: {str(e)}")
```

File: input_data.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+
+"""
+Decision trees benefit from interpretable, numerical, and diverse features. A tree splits data by testing one feature at a time. So the more relevant the features are to the task (e.g., detecting patterns in hearing loss), the better the model performs.
+"""
+
+
+def extract_features(audiogram):
```
Comment (tarnowsky) [inline input_data.py:26]: #### Consider adding error handling

```suggestion
def extract_features(audiogram):
    if len(audiogram) != 14:
        raise ValueError(f"Audiogram must have exactly 14 values (7 per ear), got {len(audiogram)}")
```

File: model_train.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import mean_squared_error
+from sklearn.tree import plot_tree
+import matplotlib.pyplot as plt
+
+
+def train_model(X, y, model_path="model.pkl"):
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, test_size=0.2, random_state=42
+    )
```
Comment (tarnowsky) [inline model_train.py]: #### Consider making the split ratio and random state configurable parameters.

```suggestion
if X is None or y is None:
        raise ValueError("Input features X and target y cannot be None")
    if len(X) != len(y):
        raise ValueError("Features X and targets y must have the same number of samples")
    
    X_train, X_test, y_train, y_test = train_test_split(
        X, y, test_size=test_size, random_state=random_state
    )
```

File: model_train.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    # TODO: Test other depth levels.
+    model = MultiOutputRegressor(DecisionTreeRegressor(max_depth=3))
+    model.fit(X_train, y_train)
+
+    y_pred = model.predict(X_test)
+    mse = mean_squared_error(y_test, y_pred)
+    print(f"Test MSE: {mse:.3f}")
+
+    joblib.dump(model, model_path)
+    print(f"Model saved to {model_path}")
```
Comment (tarnowsky) [inline model_train.py]: #### Consider adding error handling (file saving)
```suggestion
    try:
        joblib.dump(model, model_path)
        print(f"Model saved to {model_path}")
    except Exception as e:
        print(f"Error saving model to {model_path}: {e}")
        raise
```

File: predict_and_plot.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    "wysokoczęstotliwościowy P",
+    "niskoczęstotliwościowy L",
+    "niskoczęstotliwościowy P",
+]
+
+
+def predict_sample(sample, model_path="model.pkl"):
+    model = joblib.load(model_path)
+    features = extract_features(sample)
+    prediction = model.predict([features])[0]
```
Comment (tarnowsky) [inline predict_and_plot.py]: #### Consider error handling and clear error messages
```suggestion
    try:
        model = joblib.load(model_path)
    except FileNotFoundError:
        raise FileNotFoundError(f"Model file not found: {model_path}")
    except Exception as e:
        raise RuntimeError(f"Error loading model: {e}")
    
    try:
        features = extract_features(sample)
    except Exception as e:
        raise ValueError(f"Error extracting features from sample: {e}")
    
    try:
        prediction = model.predict([features])[0]
    except Exception as e:
        raise RuntimeError(f"Error making prediction: {e}")
```

**Review by tarnowsky — CHANGES_REQUESTED** [review]
Added error handling suggestion for future development.

**Review by tarnowsky — APPROVED** [review]
